app/code/Codifi/CustomerRequest/Model/ExportCsv.php

```diff
@@ -0,0 +1,175 @@
+ */
+
+namespace Codifi\CustomerRequest\Model;
+
+use Magento\Framework\Filesystem;
+use Codifi\Training\Model\NoteRepository;
+use Magento\Framework\File\Csv;
+use Magento\Framework\App\Filesystem\DirectoryList;
+use Magento\Framework\Api\FilterBuilder;
+use Magento\Framework\Api\Search\SearchCriteriaBuilder;
+use Magento\Framework\Stdlib\DateTime\TimezoneInterface;
+use Magento\Framework\Exception\FileSystemException;
+use Magento\Framework\Exception\NoSuchEntityException;
+
+/**
+ * Class ExportCsv
+ * @package Codifi\CustomerRequest\Model
+ */
+class ExportCsv
+{
+    /**
+     * Filesystem
+     *
+     * @var Filesystem
+     */
+    protected $fileSystem;
+
+    /**
+     * Note repository
+     *
+     * @var NoteRepository
+     */
+    private $noteRepository;
+
+    /**
+     * Csv processor
+     *
+     * @var Csv
+     */
+    private $csvProcessor;
+
+    /**
+     * Directory list
+     *
+     * @var DirectoryList
+     */
+    private $directoryList;
+
+    /**
+     * Filter builder
+     *
+     * @var FilterBuilder
+     */
+    private $filterBuilder;
+
+    /**
+     * Search criteria builder
+     *
+     * @var SearchCriteriaBuilder
+     */
+    private $searchCriteriaBuilder;
+
+    /**
+     * TimezoneInterface
+     *
+     * @var TimezoneInterface
+     */
+    private $timeZoneInterface;
+
+    /**
+     * New directory
+     */
+    private $newDirectory;
+
+    /**
+     * ExportCsv constructor.
+     *
+     * @param Filesystem $fileSystem
+     * @param NoteRepository $noteRepository
+     * @param Csv $csvProcessor
+     * @param DirectoryList $directoryList
+     * @param FilterBuilder $filterBuilder
+     * @param SearchCriteriaBuilder $searchCriteriaBuilder
+     * @param TimezoneInterface $timeZoneInterface
+     * @throws FileSystemException
+     */
+    public function __construct(
+        Filesystem $fileSystem,
+        NoteRepository $noteRepository,
+        Csv $csvProcessor,
+        DirectoryList $directoryList,
+        FilterBuilder $filterBuilder,
+        SearchCriteriaBuilder $searchCriteriaBuilder,
+        TimezoneInterface $timeZoneInterface
+    ) {
+        $this->noteRepository = $noteRepository;
+        $this->csvProcessor = $csvProcessor;
+        $this->directoryList = $directoryList;
+        $this->filterBuilder = $filterBuilder;
+        $this->searchCriteriaBuilder = $searchCriteriaBuilder;
+        $this->timeZoneInterface = $timeZoneInterface->date();
+        $this->newDirectory = $fileSystem->getDirectoryWrite(DirectoryList::VAR_DIR);
+    }
+
+    /**
+     * Pack customer notes to archive and delete from table
+     *
+     * @param int $period
+     * @return string
+     * @throws NoSuchEntityException
+     */
+    public function export(int $period): string
+    {
+        $currentDateTime = $this->timeZoneInterface->format('Y_m_d H:m:s');
+
+        $date = strtotime($currentDateTime . "-1 $period");
+        $newDate = date('Y-m-d H:m:s', $date);
```

Comment:
You can use Magento wrapper for php functions
Magento\Framework\Stdlib\DateTime\DateTime